examples/gsm8k/gsm8k.py

```diff
@@ -0,0 +1,292 @@
+import os
+import re
+import logging
+import datetime
+import json
+from pathlib import Path
+from typing import Dict, List, Callable, Union
+from graph_of_thoughts import controller, language_models, operations, prompter, parser
+
+
+def strip_int_result(text: str) -> int:
+        match = re.search(r'#### (\d+)', text)
+        if match:
+            return int(match.group(1))
+        return None
+
+
+def test_answer(state: Dict) -> bool:
+    """
+    Function to test whether the final solution matches ground truth.
+
+    :param state: Thought state that represents the final solution.
+    :type state: Dict
+    :return: Returns whether the solution matches the ground truth.
+    :rtype: bool
+    """
+
+    try:
+        ground_truth = state["ground_truth"]
+        current_answer = state["current"]
+        return ground_truth == current_answer
+    except:
+        return False
+
+
+class GSM8KPrompter(prompter.Prompter):
+    """
+    GSM8KPrompter provides the generation of prompts specific to the gsm8k example for the language models.
+
+    Inherits from the Prompter class and implements its abstract methods.
+    """
+
+    answer_prompt = """<Instruction> Solve the following math problems and provide the full reasoning in the answer as well as the integer solution behind ####. </Instruction>
+
+    <Examples>
+    Input: Natalia sold clips to 48 of her friends in April, and then she sold half as many clips in May. How many clips did Natalia sell altogether in April and May?
```

Comment:
Hey, isn't this basically COT and not IO, or am I mistaken?

Reply:
I think this would actually be ReAct. But you are right. This is not directly IO. CoT would be to separate it into multiple thoughts in a row. Will think about it and change it accordingly.

Reply:
You are right it is not react but a form of CoT with all thoughts in one output. I will call it single_thought for now

Reply:
makes a big difference btw (io vs single_thought)

Reply:
@fjfricke all the classic CoT papers i saw did have all the thoughts in one output. That's why its confusing, as CoT is very different from ToT or GoT in that matter.